.github/workflows/pr_test_helm_chart.yml

```
@@ -19,6 +19,27 @@ env:
   GO_VERSION: "1.23"
 
 jobs:
+  test:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: true
+
+      - name: Run Unit Tests
+        working-directory: ./kof-operator
+        run: make test
+
+
   test-and-deploy:
```

Now that you've moved `make test` to a new `test` job, please rename `test-and-deploy` to just `deploy`:
```suggestion
  deploy:
```